mysql-test/suite/galera/t/galera_password_history.test

```diff
@@ -0,0 +1,46 @@
+#
+# Test if replicated events which add rows to mysql.password_history table does not cause PK collision in this table.
+# The timestamp of the event is the part of PK. If replicated events miss usec part of the timestamp and they occur
+# on the same second, they will cause PK collision.
+# This test executes queries in bursts, so at list part of the burst should happen in the same second causing the collision.
+#
+ 
+--source include/galera_cluster.inc
+
+--echo This test does not produce the output. It tries to trigger node crash after PK collision in mysql.password_history table.
+--disable_query_log
+
+--let $count = 100
+# As the user exists it will produce series of 'Note	3163	Authorization ID 'test1'@'%' already exists.'. Disable it.
+--disable_warnings
+while ($count) {
+  CREATE USER IF NOT EXISTS 'test1'@'%' IDENTIFIED WITH 'mysql_native_password' AS '*6C387FC3893DBA1E3BA155E74754DA6682D04747';
+  --dec $count
+}
+--enable_warnings
+
+--let $count = 100
+while ($count) {
+  ALTER USER 'test1'@'%' IDENTIFIED WITH 'mysql_native_password' AS '*6C387FC3893DBA1E3BA155E74754DA6682D04747';
+  --dec $count
+}
+
+SET PASSWORD FOR 'test1'@'%' = '*6C387FC38kzfldhf8daf8fad0adf0ad0f0adf0ad';
+SET PASSWORD FOR 'test1'@'%' = '*6C387FC38kdfad9fad0fad90fa0df8ad0f8a0d9f';
+SET PASSWORD FOR 'test1'@'%' = '*6C387FC38kdf0a8s0d9f80adf808666f6fdadfad';
+SET PASSWORD FOR 'test1'@'%' = '*6C387FC38kzfld77969696969698698690888daf';
+SET PASSWORD FOR 'test1'@'%' = '*6C38fds8s8df8d88sf8d88sf8d8s8f8d8f8s8d8f';
+
+# check if mysql.password_history table was really used
+--let $assert_text = 'node_1: There should be 10 rows in mysql.password_history table.'
+--let $assert_cond = [SELECT COUNT(*) = 10 FROM mysql.password_history WHERE User = "test1"] = 1
+--source include/assert.inc
```

Comment:
Can we validate that the contents are also same
```
--let $galera_diff_statement = SELECT * FROM mysql.password_history WHERE User = "test1";
--source include/galera_diff.inc
```

Reply:
Yup, we can do this, but only for Host/User/Timestamp part. Password is kept as the hash and is unique to the server.